File: src/Azdo.Core/Metrics/Paths.cs
namespace Azdo.Core.Metrics;

/// <summary>Resolves the base directory and standard file locations for metrics persistence.</summary>
public static class MetricsPaths
{
    /// <summary>
    /// Base directory for metrics persistence files (snapshots, sprint selection,
    /// backfill marker). AZDO_CONFIG_DIR, when set, wins outright (demo mode points
    /// it at a temp directory). Otherwise the standard ~/.config/azdo-tui location.
    /// </summary>
    public static string ConfigDir()
    {
        var dir = Environment.GetEnvironmentVariable("AZDO_CONFIG_DIR");
        if (!string.IsNullOrEmpty(dir))
            return dir;
        var home = Environment.GetFolderPath(Environment.SpecialFolder.UserProfile);
        return Path.Combine(home, ".config", "azdo-tui");
    }

    /// <summary>Standard snapshot file location (metrics.jsonl).</summary>
    public static string DefaultSnapshotPath() => Path.Combine(ConfigDir(), "metrics.jsonl");

    /// <summary>Standard trend-view sprint selection location (metrics-selection.json).</summary>
    public static string DefaultSelectionPath() => Path.Combine(ConfigDir(), "metrics-selection.json");

    /// <summary>Standard backfill marker location (.metrics-backfill-done).</summary>
    public static string DefaultBackfillMarkerPath() => Path.Combine(ConfigDir(), ".metrics-backfill-done");

    /// <summary>Creates the directory holding the snapshot file if it doesn't exist. Safe to repeat.</summary>
    public static void EnsureSnapshotDir(string path)
    {
        var dir = Path.GetDirectoryName(path);
        if (!string.IsNullOrEmpty(dir))
            Directory.CreateDirectory(dir);
    }
}

File: reference/internal/metrics/paths.go
package metrics

import (
	"fmt"
	"os"
	"path/filepath"
)

// configDir returns the base directory for metrics persistence files
// (snapshots, sprint selection, backfill marker).
//
// AZDO_CONFIG_DIR, when set, wins outright — demo mode points it at a temp
// directory so showcasing the metrics tab never reads or writes the user's
// real history. Otherwise the standard ~/.config/azdo-tui location is used,
// preserving existing behavior for everyone else.
func configDir() (string, error) {
	if dir := os.Getenv("AZDO_CONFIG_DIR"); dir != "" {
		return dir, nil
	}
	home, err := os.UserHomeDir()
	if err != nil {
		return "", fmt.Errorf("home dir: %w", err)
	}
	return filepath.Join(home, ".config", "azdo-tui"), nil
}
